azure/services/subnets/spec.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subnets
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+// SubnetSpec defines the specification for a Subnet.
+type SubnetSpec struct {
+	Name              string
+	ResourceGroup     string
+	SubscriptionID    string
+	CIDRs             []string
+	VNetName          string
+	VNetResourceGroup string
+	IsVNetManaged     bool
+	RouteTableName    string
+	SecurityGroupName string
+	Role              infrav1.SubnetRole
+	NatGatewayName    string
+}
+
+// ResourceName returns the name of the subnet.
+func (s *SubnetSpec) ResourceName() string {
+	return s.Name
+}
+
+// ResourceGroupName returns the name of the vnet's resource group.
```

Comment:
```suggestion
// ResourceGroupName returns the name of the vnet's resource group which the subnet is part of.
```

? 

this might not be the best way of saying it but essentially want to make it clear that we're intentionally using the vnet RG so it doesn't seem like a bug to someone reading this code for the first time